### PR TITLE
Reworked GraphQL tutorial

### PR DESCRIPTION
**Pull request by dobooth**
## Purpose of this pull request

This pull request (PR) enhances and fixes the GraphQL tutorial.
It clarifies steps, adds instructions for creating a coupon and other markup fixes.

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/graphql/tutorials/checkout/index.html

Reworked the [GraphQL tutorial](https://devdocs.magento.com/guides/v2.4/graphql/tutorials/checkout/index.html
).

### CODE REVIEW

**Review by keharper — CHANGES_REQUESTED** [review]

File: src/guides/v2.3/graphql/tutorials/checkout/index.md [hunk 1]
```diff
@@ -16,31 +16,33 @@ contributor_link: https://www.atwix.com/
 
 This tutorial describes how to place an order through GraphQl. Customers can make purchases in two ways:
 
-*  As a logged-in user
-*  As a guest user who does not create an account
+-  As a logged-in user
+-  As a guest user who does not create an account
 
-The **10-step tutorial** generally takes **30 minutes**.
+The 10-step tutorial generally takes 30 minutes.
```
[the remark this answers is not in the document]
Comment (dobooth) [reply src/guides/v2.3/graphql/tutorials/checkout/index.md]: Done

File: src/guides/v2.3/graphql/tutorials/checkout/index.md [hunk 2]
```diff
@@ -16,31 +16,33 @@ contributor_link: https://www.atwix.com/
...
 
 Complete the following prerequisites:
 
-*  Install a Magento 2.3.2 instance with sample data.
+-  Install a Magento 2 instance with sample data.
 
   The sample data defines a functional store, called Luma, that sells fitness clothing and accessories. The store does not provide any sandbox accounts for testing credit card payments, so transactions will be simulated using an offline [payment method](https://glossary.magento.com/payment-method).
 
-*  Install a GraphQl client. You can use any GraphQl client to send calls to Magento. [GraphiQL](https://electronjs.org/apps/graphiql) is recommended.
+-  In the Magento admin, create a coupon that will be used in [Step 7. Apply a coupon]({{page.baseurl}}/graphql/tutorials/checkout/checkout-coupon.html).
```
[the remark this answers is not in the document]
Comment (dobooth) [reply src/guides/v2.3/graphql/tutorials/checkout/index.md]: Done.

File: src/guides/v2.3/graphql/tutorials/checkout/checkout-customer.md [hunk 3]
```diff
@@ -88,8 +88,18 @@ mutation {
 }
 ```
 
-[Authorization tokens]({{page.baseurl}}/graphql/authorization-tokens.html) describes the mutation further.
-
 ## Verify this step {#verify-step}
 
 Sign in as a customer to the website using the email `john.doe@example.com` and password `b1b2b3l@w+`. You should be successfully logged in.
+
+## Add Authorization header
```
[the remark this answers is not in the document]
Comment (dobooth) [reply src/guides/v2.3/graphql/tutorials/checkout/checkout-customer.md:91]: Done

File: src/guides/v2.3/graphql/tutorials/checkout/checkout-billing-address.md [hunk 4]
```diff
@@ -220,17 +205,18 @@ mutation {
         ]
       }
     }
-  }
-}
```
[the remark this answers is not in the document]
Comment (dobooth) [reply src/guides/v2.3/graphql/tutorials/checkout/checkout-billing-address.md:224]: Done

File: src/guides/v2.3/graphql/tutorials/checkout/checkout-place-order.md [hunk 5]
```diff
@@ -18,12 +18,11 @@ The `placeOrder` mutation places an order.
...
 
-**Request:**
+Send the customer's authorization token in the `Authorization` parameter of the header. See [Authorization tokens]({{page.baseurl}}/graphql/authorization-tokens.html) for more information.
 
-{:.bs-callout .bs-callout-info}
-For logged-in customers, send the customer's authorization token in the `Authorization` parameter of the header. See [Authorization tokens]({{page.baseurl}}/graphql/authorization-tokens.html) for more information.
+**Request:**
 
-```text
+```graphql
```
Comment (keharper) [inline src/guides/v2.3/graphql/tutorials/checkout/checkout-place-order.md:25]: In the last line of the topic, the right parenthesis is missing.

**Review by keharper — CHANGES_REQUESTED** [review]

File: src/guides/v2.3/graphql/tutorials/checkout/checkout-coupon.md [hunk 6]
```diff
@@ -20,12 +20,41 @@ Use [applyCouponToCart]({{ page.baseurl }}/graphql/mutations/apply-coupon.html)
...
+## Create a coupon
+
+Coupons must be generated from the Admin.
+
+Creating a coupon is described in [Coupon Codes](https://docs.magento.com/user-guide/marketing/price-rules-cart-coupon.html).
+For the purpose of this tutorial, create a Cart Price Rule with:
+
+For **Rule Information**:
+
+-  Rule Name: Watch Coupon
```
[the remark this answers is not in the document]
Comment (dobooth) [reply src/guides/v2.3/graphql/tutorials/checkout/checkout-coupon.md:32]: Done

File: src/guides/v2.3/graphql/tutorials/checkout/checkout-place-order.md (hunk 5 above)
Comment (dobooth) [reply src/guides/v2.3/graphql/tutorials/checkout/checkout-place-order.md:25]: Done.

File: src/guides/v2.3/graphql/tutorials/checkout/checkout-billing-address.md [hunk 7]
```diff
@@ -25,53 +25,42 @@ Use the [setBillingAddressOnCart]({{ page.baseurl }}/graphql/mutations/set-billi
 
 ## Add a new billing address
 
-The following mutation adds a new billing address. `{ CART_ID }` is the unique shopping cart ID from [Step 2. Create empty cart]({{ page.baseurl }}/graphql/tutorials/checkout/checkout-add-product-to-cart.html).
+Similar to the previous step, we will add a billing address to the customer. `{ CART_ID }` is the unique shopping cart ID from [Step 2. Create empty cart]({{ page.baseurl }}/graphql/tutorials/checkout/checkout-add-product-to-cart.html). In this example, the `default_billing` parameter is set to `true`. The street address is also different, so we can see that different addresses are being created.
```
Comment (keharper) [inline src/guides/v2.3/graphql/tutorials/checkout/checkout-billing-address.md:28]: Get rid of everything but adding a billing address to the cart, so the procedure is the same for guests and customers. 

The shipping and billing addresses are now different. That's OK, but be sure to mention that.

File: src/guides/v2.3/graphql/tutorials/checkout/checkout-shipping-address.md [hunk 8]
```diff
@@ -14,28 +14,23 @@ contributor_name: Atwix
...
-{:.bs-callout .bs-callout-info}
-For logged-in customers, send the customer's authorization token in the `Authorization` parameter of the header. See [Authorization tokens]({{page.baseurl}}/graphql/authorization-tokens.html) for more information.
-
-```text
+```graphql
 mutation {
   setShippingAddressesOnCart(
     input: {
-      cart_id: "{ CART_ID }"
+      cart_id: "{{ CART_ID }}"
```
[the remark this answers is not in the document]
Comment (dobooth) [reply src/guides/v2.3/graphql/tutorials/checkout/checkout-shipping-address.md:33]: That's from trying to get variables working in Altair.

File: src/guides/v2.4/graphql/tutorials/checkout/checkout-shipping-method.md [hunk 9]
```diff
@@ -22,21 +22,22 @@ The `setShippingMethodsOnCart` mutation defines the delivery methods for your or
 
 `{ CART_ID }` is the unique shopping cart ID from [Step 2. Create empty cart]({{ page.baseurl }}/graphql/tutorials/checkout/checkout-add-product-to-cart.html).
 
-{:.bs-callout .bs-callout-info}
+The `carrier_code` and `method_code` values come from the results of the mutation on the previous step.
```
Comment (keharper) [inline src/guides/v2.4/graphql/tutorials/checkout/checkout-shipping-method.md:25]: It's currently missing from the tutorial. This should link to the shipping address step

File: src/guides/v2.4/graphql/tutorials/checkout/checkout-shipping-method.md [hunk 10]
```diff
@@ -22,21 +22,22 @@ The `setShippingMethodsOnCart` mutation defines the delivery methods for your or
...
+```graphql
 mutation {
   setShippingMethodsOnCart(input: {
     cart_id: "{ CART_ID }"
     shipping_methods: [
       {
-        carrier_code: "ups"
-        method_code: "GND"
+        carrier_code: "tablerate"
+          method_code: "bestway"
```
Comment (keharper) [inline src/guides/v2.4/graphql/tutorials/checkout/checkout-shipping-method.md:40]: The alignment is off here

**Review by keharper — COMMENTED** [review]
A couple of nits, but the setShippingAddressOnCart response needs to be expanded.

File: src/guides/v2.3/graphql/tutorials/checkout/checkout-shipping-address.md [hunk 11]
```diff
@@ -14,24 +14,19 @@ contributor_name: Atwix
...
+## Add shipping address to the cart
 
-## Create a new shipping address
+In this step, we use the `setShippingAddressesOnCart` mutation to add a shipping address to the cart.
 
-The following mutation adds a shipping address to the quote.
+If using guest checkout, run the following example.
 
-`{ CART_ID }` is the unique shopping cart ID from [Step 2. Create empty cart]({{ page.baseurl }}/graphql/tutorials/checkout/checkout-add-product-to-cart.html).
+If using a logged in customer, send the customer's authorization token in the `Authorization` parameter of the header. See [Authorization tokens]({{page.baseurl}}/graphql/authorization-tokens.html) for more information.
```
[the remark this answers is not in the document]
Comment (dobooth) [reply src/guides/v2.3/graphql/tutorials/checkout/checkout-shipping-address.md:25]: Done

File: src/guides/v2.4/graphql/tutorials/checkout/checkout-shipping-method.md (hunk 9 above)
Comment (dobooth) [reply src/guides/v2.4/graphql/tutorials/checkout/checkout-shipping-method.md:25]: Done

File: src/guides/v2.4/graphql/tutorials/checkout/checkout-shipping-method.md (hunk 10 above)
Comment (dobooth) [reply src/guides/v2.4/graphql/tutorials/checkout/checkout-shipping-method.md:40]: Got it.

**Review by keharper — APPROVED** [review]

**Comment by keharper** [comment]
running tests

**Pull request merged**